app/src/main/java/io/novafoundation/nova/app/root/presentation/deepLinks/DAppDeepLinkHandler.kt

```diff
@@ -0,0 +1,48 @@
+package io.novafoundation.nova.app.root.presentation.deepLinks
+
+import android.net.Uri
+import io.novafoundation.nova.common.utils.Urls
+import io.novafoundation.nova.feature_account_api.domain.interfaces.AccountRepository
+import io.novafoundation.nova.feature_dapp_api.data.model.DappMetadata
+import io.novafoundation.nova.feature_dapp_api.data.repository.DAppMetadataRepository
+import io.novafoundation.nova.feature_dapp_impl.DAppRouter
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.emptyFlow
+
+private const val DAPP_DEEP_LINK_PREFIX = "/open/dapp"
+
+class DAppDeepLinkHandler(
+    private val accountRepository: AccountRepository,
+    private val dappRepository: DAppMetadataRepository,
+    private val dAppRouter: DAppRouter
+) : DeepLinkHandler {
+
+    override val callbackFlow: Flow<CallbackEvent> = emptyFlow()
+
+    override suspend fun matches(data: Uri): Boolean {
+        val userAccounts = accountRepository.allMetaAccounts()
+        val path = data.path ?: return false
+        val url = data.getDappUrl() ?: return false
+        val dappInfo = syncAndGetDapp(url)
```

Comment:
Similarily to previous PR, I suggest `matchers` to only validate the path, and perform the rest in `handleDeepLink`